add onclick hook to rui-card-create

ajkitson opened the pull request:
This PR allows you to hook into rui-create-card so that when it is clicked, instead of flipping over and adding a name, you can do something else, like launch a modal. Added so that we can launch a modal form on creating a source or destination where you can enter your environment type.

Also cleaned up some formatting.

jerseygooner commented:
Can we update _all.html so that there is an example of the function ability in the docs? Something simple like an `alert()` should do the trick.

jerseygooner commented on src/js/directives/card-create.js:
```diff
@@ -1,18 +1,26 @@
-      $scope.create = function(){
-        $scope.createFn({name: $scope.createinput});
-        $scope.editing = false;
+    link: function(scope, element, attrs) {
```
It looks like the `createFn` will not be available in the component if the `clickFn` attribute is passed in with a value. 

It would be nice to either have documentation clarifying this behavior, or somehow allow the `clickFn` to use the `createFn`.

ajkitson commented:
Sounds good. Added an example to _all.html and refactored the click function so it could be used in conjunction with the create function... and is generally a bit cleaner. Also included dst.

ajkitson commented:
Actually, now that I'm working with my actual use case, I want to handle the onclick a little differently. If you're using the click function, it's because you intend to hijack the record creation process. Removed the stuff where you could have a click function and then still get the name input.

jerseygooner commented:
The value in createInput is 'sticky' in that we don't clear it out upon `onClick()`. I guess it could work either way, but it was weird to me when I clicked the card, typed something, then clicked cancel, then clicked on the card again and my text was still there.

ajkitson commented:
Yeah, it's always been like that--shouldn't be any change with this PR. 

I'm undecided on whether this is a feature or something else :)  FWIW, once we switch to environment type selection on source/dest creation, the only place that will still use the non-modal version is the organization creation, which is admin only at the moment

jerseygooner commented:
Not sure if this is how we'd want it to work, but as you mention, it is unlikely this will come up in the current applications of this directive. 

This looks good to go, but there are merge conflicts.

ajkitson commented:
Alrighty, was just a merge conflict with the dst/ files.
